project-base/storefront/components/Pages/Order/TransportAndPayment/TransportAndPaymentContent.tsx

```diff
@@ -120,58 +95,122 @@ export const TransportAndPaymentContent: FC<TransportAndPaymentContentProps> = (
         router.push(contactInformationUrl);
     };
 
-    const lastOrderPickupPlace: ListedStoreFragmentApi | null = useMemo(() => {
-        if (lastOrder?.pickupPlaceIdentifier === undefined || lastOrder.pickupPlaceIdentifier === null) {
-            return null;
-        }
-
-        if (packeteryPickupPoint?.identifier === lastOrder.pickupPlaceIdentifier) {
-            return packeteryPickupPoint;
-        }
-
-        if (pickupPlaceData?.store !== undefined && pickupPlaceData.store !== null) {
-            return getGtmPickupPlaceFromStore(lastOrder.pickupPlaceIdentifier, pickupPlaceData.store);
-        }
-
-        return getGtmPickupPlaceFromLastOrder(lastOrder.pickupPlaceIdentifier, lastOrder);
-    }, [lastOrder, packeteryPickupPoint, pickupPlaceData?.store]);
+    useEffect(() => {
+        setIsLoadingInitialTransportAndPaymentData((!lastOrderData && isUserLoggedIn) || !transportsData);
+    }, [lastOrderData, isUserLoggedIn, transportsData]);
 
     return (
         <>
-            {transports !== undefined && transports.length > 0 && (
-                <TransportAndPaymentSelect
-                    transports={transports}
-                    lastOrderPickupPlace={lastOrderPickupPlace}
-                    lastOrderTransportUuid={lastOrder?.transport.uuid ?? null}
-                    lastOrderPaymentUuid={lastOrder?.payment.uuid ?? null}
-                    changeTransportInCart={changeTransportInCart}
-                    changePaymentInCart={changePaymentInCart}
-                    isTransportSelectionLoading={isTransportSelectionLoading}
-                />
-            )}
-            <OrderAction
-                buttonBack={t('Back')}
-                buttonNext={t('Contact information')}
-                hasDisabledLook={
-                    hasValidationErrors(transportAndPaymentValidationMessages) ||
-                    isTransportSelectionLoading ||
-                    isPaymentSelectionLoading
-                }
-                isLoading={
-                    (isTransportSelectionLoading || isPaymentSelectionLoading) && transport !== null && payment !== null
-                }
-                withGapTop
-                withGapBottom
-                buttonBackLink={cartUrl}
-                nextStepClickHandler={onSelectTransportAndPaymentHandler}
-            />
-            {isErrorPopupVisible && (
-                <ErrorPopup
-                    onCloseCallback={() => setErrorPopupVisibility(false)}
-                    fields={transportAndPaymentValidationMessages}
-                    gtmMessageOrigin={GtmMessageOriginType.transport_and_payment_page}
-                />
-            )}
+            <OrderLayout
+                activeStep={2}
+                isTransportOrPaymentLoading={Boolean(isTransportSelectionLoading) || isPaymentSelectionLoading}
```

Comment:
I know it's not your change, but can we remove Boolean? Since it's already boolean

```suggestion
                isTransportOrPaymentLoading={isTransportSelectionLoading || isPaymentSelectionLoading}
```

Reply:
Yep, changed...